ERCS/erc-4337.md

```diff
@@ -118,19 +125,20 @@ interface IAccount {
 }
 ```
 
-The `userOpHash` is a hash over the userOp (except signature), entryPoint and chainId.  
+The `userOpHash` is a hash over the `userOp` (except `signature`), `entryPoint` and `chainId`.
 
 The account:
 
-* MUST validate the caller is a trusted EntryPoint
+* MUST validate the caller is a trusted `EntryPoint`
 * MUST validate that the signature is a valid signature of the `userOpHash`, and
   SHOULD return SIG_VALIDATION_FAILED (and not revert) on signature mismatch. Any other error MUST revert.
-* MUST pay the entryPoint (caller) at least the "missingAccountFunds" (which might be zero, in case the current account's deposit is high enough)
+* SHOULD not return early when returning `SIG_VALIDATION_FAILED`. Instead, it SHOULD complete the normal flow to enable performing a gas estimation for the validation function.
+* MUST pay the `EntryPoint` (caller) at least the `missingAccountFunds` (which might be zero, in case the current account's deposit is high enough)
```

Comment:
```suggestion
* MUST pay the `EntryPoint` (caller) at least the `missingAccountFunds` (which might be zero, in case the current account's deposit is sufficient)
```